app/view-plants/page.tsx

```diff
@@ -1,58 +1,77 @@
 'use client';
 
 import { useEffect, useState } from 'react';
+import { useRouter } from 'next/navigation';
 import { UUID } from 'crypto';
-import supabase from '@/api/supabase/createClient';
-import { getAllPlants, getPlantById } from '@/api/supabase/queries/plants';
+import {
+  getAllPlants,
+  getMatchingPlantForUserPlant,
+} from '@/api/supabase/queries/plants';
+import { getUserPlantsByUserId } from '@/api/supabase/queries/userPlants';
 import PlantCard from '@/components/PlantCard';
-import { Plant } from '@/types/schema';
+import { OwnedPlant, Plant } from '@/types/schema';
 
 export default function Page() {
+  const router = useRouter();
   const [viewingOption, setViewingOption] = useState<'myPlants' | 'all'>(
     'myPlants',
   );
   const [inAddMode, setInAddMode] = useState<boolean>(false);
-
   const [plants, setPlants] = useState<Plant[]>([]);
-  const [userPlants, setUserPlants] = useState<Plant[]>([]);
+  const [selectedPlants, setSelectedPlants] = useState<Plant[]>([]);
+  const [ownedPlants, setOwnedPlants] = useState<OwnedPlant[]>([]);
+
   const user_id: UUID = 'e72af66d-7aae-45f6-935a-187197749d9f';
   const userState = 'TENNESSEE';
-  async function fetchUserPlants(user_id: UUID) {
-    const { data, error } = await supabase
-      .from('user_plants')
-      .select('plant_id')
-      .eq('user_id', user_id)
-      .is('date_harvested', null);
 
-    if (error) {
-      console.error('Error fetching plant IDs:', error);
-      return [];
-    }
-    if (!data) return [];
-    const plantIds = data.map(row => row.plant_id) || [];
+  async function fetchUserPlants(user_id: UUID) {
+    const fetchedUserPlants = await getUserPlantsByUserId(user_id);
 
-    const plantsUser: Plant[] = await Promise.all(
-      plantIds.map(plantId => getPlantById(plantId)),
+    const ownedPlants: OwnedPlant[] = await Promise.all(
+      fetchedUserPlants.map(async userPlant => {
+        const plant = await getMatchingPlantForUserPlant(userPlant);
+        return {
+          userPlantId: userPlant.id,
+          plant,
+        };
+      }),
     );
-    return plantsUser;
+
+    return ownedPlants;
   }
+
   useEffect(() => {
-    const fetchPlantSeasonality = async () => {
+    (async () => {
       const plantList = await getAllPlants();
       const result = plantList.filter(plant => plant.us_state === userState);
       setPlants(result);
-    };
-
-    fetchPlantSeasonality();
+    })();
   }, []);
+
   useEffect(() => {
     const fetchData = async () => {
```

Comment:
can we directly just move fetchUserPlants into the useEffect instead of using fetchData as another wrapper?

so it could look like 
```
useEffect(() => {
  const fetchUserPlants = async () => {
    const fetchedUserPlants = await getUserPlantsByUserId(user_id);

    const ownedPlants: OwnedPlant[] = await Promise.all(
      fetchedUserPlants.map(async userPlant => {
        const plant = await getMatchingPlantForUserPlant(userPlant);
        return {
          userPlantId: userPlant.id,
          plant,
        };
      }),
    );

    setOwnedPlants(ownedPlants);
  }, []); // eventually when ProfileContext is connected, userId would be in the dependency list.
```

See [IJP](https://github.com/calblueprint/immigration-justice-project/blob/main/src/app/language-support/page.tsx) for an example of using queries in the useEffect

Reply:
note: since the queries (both `getUserPlantsByUserId` and `getMatchingPlantForUserPlant`) throw an error on error, we may want to use a try / catch block here and wherever a query is called to handle the error, i.e. console.error() it in the future (see IJP's example above)